src/components/document-id/DocumentId.vue

```diff
@@ -0,0 +1,256 @@
+  documentId = ''
+  isVerifyingDocId = false
+  isVerifiedDocId = false
+  generateDocumentId = false
+  docIdError = []
+
+  /** Getter for the Document ID error styling */
+  get showBorderError (): boolean {
+    return this.validate && !this.isValidDocumentId
+  }
+
+  /** Getter for the Document ID validation state */
+  get isValidDocumentId (): boolean {
+    return this.isVerifiedDocId || this.generateDocumentId
+  }
+
+  /**
+   * Validates the Document ID by making an API call to the document services.
+   * Constructs the URL and headers for the API call, including authorization and account information.
+   * If the API call returns status 200, the Document ID is not unique.
+   * If the API call returns status 400, the Document ID failed the checksum format.
+   * If the API call returns status 404, the Document ID is unique and valid.
+   * If the API call returns any other status, the document ID is not valid.
+   * *
+   * @returns {Promise<number>} - The status code of the API response.
+   */
+  async fetchDocumentIdStatus (): Promise<number> {
+    // Construct the URL for the API call
+    const url = `${this.docApiUrl}/documents/verify/${this.documentId}`
+
+    // Set up the headers for the API call
+    const config = { headers: {
+      'Authorization': `Bearer ${sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)}`,
+      'x-apikey': this.docApiKey }
+    }
+
+    // Add the Account-Id header if the current account is available
+    const currentAccount = sessionStorage.getItem(SessionStorageKeys.CurrentAccount)
+    if (currentAccount) {
+      const accountInfo = JSON.parse(currentAccount)
+      config.headers['Account-Id'] = accountInfo.id
+    }
+
+    try {
+      // Make the API call to validate the Document ID. If the call is successful(200), the Document ID is not unique
+      const test = await axios.get(url, config)
+      return test.status
+    } catch (error: any) {
+      // If the API call fails, return the status code
+      if ([400, 404].includes(error.response.status)) {
+        return error.response.status
+      }
+    }
+  }
+
+  /**
+   * Validates the Document ID if its length is 8 characters.
+   * Depending on the API response, it sets the appropriate error messages and verification status.
+   * If the Document ID is verified, it emits the Document ID.
+   */
+  @Watch('documentId')
+  async onDocumentIdComplete (): Promise<void> {
+    // Validate the document id if the length is 8 characters
+    if (this.documentId?.length === 8) {
+      this.isVerifyingDocId = true
+      // Validate the document ID
+      const documentIdStatus = await this.fetchDocumentIdStatus()
+      switch (documentIdStatus) {
+        case 200:
+          this.docIdError = ['Must be unique number']
+          this.isVerifiedDocId = false
+          break
+        case 400:
+          this.docIdError = ['Document ID is invalid']
+          this.isVerifiedDocId = false
+          break
+        case 404:
```

Comment:
We do have use a  library for HTTP response code elsewhere, if you want to use it.
-> http-status-codes

Reply:
Will have a look!